map/map_loader/src/lanelet2_map_loader/lanelet2_map_loader_node.cpp

```diff
@@ -78,6 +81,31 @@ void Lanelet2MapLoaderNode::on_map_projector_info(
     return;
   }
 
+  std::string format_version{"null"}, map_version{""};
+  lanelet::io_handlers::AutowareOsmParser::parseVersions(
+    lanelet2_filename, &format_version, &map_version);
+  if (format_version == "null" || format_version.empty() || !isdigit(format_version[0])) {
+    RCLCPP_WARN(
+      get_logger(),
+      "%s has no format_version(null) or non semver-style format_version(%s) information",
+      lanelet2_filename.c_str(), format_version.c_str());
+    if (!allow_unsupported_version) {
+      throw std::invalid_argument(
+        "allow_unsupported_version is false, so stop loading lanelet map");
+    }
+  } else if (const int map_major_ver = static_cast<int>(format_version[0] - '0');
```

Comment:
This code doesn't consider cases where format_version is 10 or larger.
If `format_version` is 10, then `map_major_ver` is 1.

Reply:
That's right, I will use regex.